docs/src/how-to/programmers-guide.rst

```diff
@@ -0,0 +1,202 @@
+.. meta::
+  :description: Tensile documentation and API reference
+  :keywords: Tensile, GEMM, Tensor, ROCm, API, Documentation
+.. highlight:: none
+
+.. _programmers-guide:
+
+********************************************************************
+Programmer's guide
+********************************************************************
+
+This document provides necessary information for programmers interested in contributing to the Tensile source code.
+
+.. _development-environment:
+
+=======================
+Development environment
+=======================
+
+ROCm is the base requirement for contributing to Tensile. See if ROCm is supported on your platform by verifying the `supported operating systems <https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html#supported-operating-systems>`_ list.
+Then, follow the steps given in the :ref:`installation` guide.
+
+=====================
+Developing in Docker
```

Comment:
@bstefanuk Is this section under Development environment? 
Also, is ROCm installation required if using docker?

Reply:
Yes it should be under Development  environment, typically it's best to setup a docker-based workflow for development so it's a good idea to keep it here.

ROCm should come installed in AMDs docker images.

Reply:
Done.